crt_portal/cts_forms/filters.py

```diff
@@ -12,6 +12,7 @@
 from django.http.request import QueryDict, MultiValueDict
 
 from utils.datetime_fns import change_datetime_to_end_of_day
+from .signals import get_user_section
```

Comment:
hmm, we probably shouldn't be importing utilities from signals like this.

I see we're also doing it with get_client_ip

Can you move these non-signal methods to a utility file (request_utils or something)?

[signals isn't meant to be imported until the AppConfig runs](https://docs.djangoproject.com/en/4.2/topics/signals/#django.dispatch.receiver) its `ready` callback. Importing them early might cause strange behavior, as it'll register the `@receiver` methods below too early.

Find in all files for `from.*signals` should give a full list of offending code.